File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0 ; i < nums.length; i++){
            int n = nums[i];
            int x = target - n;
            if(map.containsKey(x)) return new int[] {map.get(x) , i};
            map.put(n, i);
        }
        return null;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i, num in enumerate(nums):
            sol = target - num
            if sol in hashmap:
                return [hashmap[sol], i]
            hashmap[num] = i
        return None
        